test/rule_definition_tools.jl

```diff
@@ -0,0 +1,65 @@
+@testset "rule_definition_tools.jl" begin
+    @testset "@non_differentiable" begin
+        @testset "nondiff_2_1" begin
```

Comment:
Could you explain why this testset / function is called `nondiff_2_1`?

Reply:
2 inputs 1 output
Do i need a comment to that effect?

Reply:
Please.

Reply:
```suggestion
        @testset "two input one output function" begin
```